src/utilities/BatchTable.js

```diff
@@ -0,0 +1,67 @@
+import { BatchTableHierarchyExtension } from './BatchTableHierarchyExtension.js';
+import { FeatureTable } from './FeatureTable.js';
+
+export class BatchTable extends FeatureTable {
+
+	constructor( buffer, batchSize, start, headerLength, binLength ) {
+
+		super( buffer, start, headerLength, binLength );
+		this.batchSize = batchSize;
+
+		this.extensions = {};
+		const extensions = this.header.extensions;
+		if ( extensions ) {
+
+			if ( extensions[ '3DTILES_batch_table_hierarchy' ] ) {
+
+				this.extensions[ '3DTILES_batch_table_hierarchy' ] = new BatchTableHierarchyExtension( this );
+
+			}
+
+		}
+
+	}
+
+	getData( key, componentType = null, type = null ) {
+
+		console.warn( 'BatchTable: BatchTable.getData is deprecated. Use BatchTable.getDataFromId instead.' );
```

Comment:
Forgot about the b3dmExample that tests batch tables - [these lines](https://github.com/NASA-AMMOS/3DTilesRendererJS/blob/a232cb76591f88241abc5e67ba4b91a95bc7d549/example/b3dmExample.js#L191-L194) should be changed now that we've changed the interface.

Reply:
Well spotted ! I updated the example.